### ISSUE TITLE
Issue while building the code in mac os mojave

### ISSUE THREAD

**Issue by rkchilaka** [servo/servo#23015]
./mach build -d

Exception: Your system's GStreamer libraries are out of date (we need at least 1.12). If you're unable to install them, let us know by filing a bug!

  File "/Users/ramakrishna.chilaka/Desktop/servo/python/servo/build_commands.py", line 303, in build
    env = self.build_env(target=target, is_build=True)
  File "/Users/ramakrishna.chilaka/Desktop/servo/python/servo/command_base.py", line 612, in build_env
    if self.needs_gstreamer_env(target):
  File "/Users/ramakrishna.chilaka/Desktop/servo/python/servo/command_base.py", line 560, in needs_gstreamer_env
    install them, let us know by filing a bug!")

I installed GStreamer libraries through brew and manually downloading from GStreamer website. installed OpenSSL from brew. Still, I am not able to build. Kindly, please help, if anyone faced the same issue before.

**Comment by FateXRebirth** [comment]
same issue encounter, any solution?
I have upgraded my gstreamer version to 1.14.4 by typing "brew upgrade gstreamer"

**Comment by janczer** [comment]
I have the same problem. MacOS Mojave 10.14.3

**Comment by CYBAI** [comment]
> I have the same problem. MacOS Mojave 10.14.3

I'm on macOS Mojave 10.14.3 with `gstreamer 1.14.4` and I can build Servo successfully.

```
→ gst-inspect-1.0 --version
gst-inspect-1.0 version 1.14.4
GStreamer 1.14.4
Unknown package origin
```

**Comment by rkchilaka** [comment]
I installed gstreamer libraries freshly through brew, they are of latest version.😄

**Comment by CYBAI** [comment]
How about checking with `pkg-config`?

````
→ pkg-config --modversion gstreamer-1.0
1.14.4
```

**Comment by josephhutch** [comment]
> How about checking with pkg-config?

I'm getting 1.14.4

> and also try to run this locally with a python script

The script prints 1 with no exceptions

I'm also having this issue on MacOS 10.14.3.  Let me know if I can provide any other information to get this issue resolved.

**Comment by CYBAI** [comment]
@josephhutch I just updated the python script snippet to not use `PIPE`. Could you run the script again? Thanks!

**Comment by josephhutch** [comment]
@CYBAI Still getting 1

**Comment by CYBAI** [comment]
@josephhutch  Hmm, sorry, could you run it again? I think this would print the error 🤔

```py
import subprocess

print subprocess.check_call(["pkg-config", "gstreamer-1.0 >= 1.12"])
print subprocess.check_output(["pkg-config", "gstreamer-1.0 >= 1.12"])
```

**Comment by rkchilaka** [comment]
> @josephhutch Hmm, sorry, could you run it again? I think this would print the error 🤔
> 
> ```python
> import subprocess
> 
> print subprocess.check_call(["pkg-config", "gstreamer-1.0 >= 1.12"])
> print subprocess.check_output(["pkg-config", "gstreamer-1.0 >= 1.12"])
> ```
On running this script i got the following output

```
>>> print subprocess.check_call(["pkg-config", "gstreamer-1.0 >= 1.12"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['pkg-config', 'gstreamer-1.0 >= 1.12']' returned non-zero exit status 1
>>> print subprocess.check_output(["pkg-config", "gstreamer-1.0 >= 1.12"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['pkg-config', 'gstreamer-1.0 >= 1.12']' returned non-zero exit status 1
```

**Comment by CYBAI** [comment]
Can someone try to apply this patch and see if it will kill the exception?

```diff
diff --git a/python/servo/bootstrap.py b/python/servo/bootstrap.py
index c7060b3888..1243d6a0a3 100644
--- a/python/servo/bootstrap.py
+++ b/python/servo/bootstrap.py
@@ -46,8 +46,7 @@ def install_trusty_deps(force):
 
 
 def check_gstreamer_lib():
-    return subprocess.call(["pkg-config", "gstreamer-1.0 >= 1.12"],
-                           stdout=PIPE, stderr=PIPE) == 0
+    return subprocess.check_output('pkg-config --modversion gstreamer-1.0') >= 1.12
 
 
 def run_as_root(command, force=False):
```

**Comment by CYBAI** [comment]
Could someone try to apply this patch and see if it will pass the build? 🤔 Thanks.

```diff
diff --git a/python/servo/bootstrap.py b/python/servo/bootstrap.py
index c7060b3888..1243d6a0a3 100644
--- a/python/servo/bootstrap.py
+++ b/python/servo/bootstrap.py
@@ -46,8 +46,7 @@ def install_trusty_deps(force):
 
 
 def check_gstreamer_lib():
-    return subprocess.call(["pkg-config", "gstreamer-1.0 >= 1.12"],
-                           stdout=PIPE, stderr=PIPE) == 0
+    return subprocess.check_output('pkg-config --modversion gstreamer-1.0') >= 1.12
 
 
 def run_as_root(command, force=False):
```

**Comment by Feliix42** [comment]
For me, the build is now successfully initiated.

**Comment by josephhutch** [comment]
@CYBAI, that worked for me as well.  Thank you for the help!

@Feliix42, the libffi issue can be solved by adding the environment variable PKG_CONFIG_PATH with libffi's path.  See `brew info libffi`

**Comment by CYBAI** [comment]
@josephhutch @Feliix42 Could you help me try if this patch also works for you? Thanks!

Btw, maybe it's good to `./mach clean` to have a clean environment and build it again, thanks!

```diff
diff --git a/python/servo/bootstrap.py b/python/servo/bootstrap.py
index c7060b3888..6456f1a075 100644
--- a/python/servo/bootstrap.py
+++ b/python/servo/bootstrap.py
@@ -46,7 +46,7 @@ def install_trusty_deps(force):
 
 
 def check_gstreamer_lib():
-    return subprocess.call(["pkg-config", "gstreamer-1.0 >= 1.12"],
+    return subprocess.call(["pkg-config", "--atleast-version=1.12", "gstreamer-1.0"],
                            stdout=PIPE, stderr=PIPE) == 0
```

**Comment by Feliix42** [comment]
This patch also works, but only if I `set -gx PKG_CONFIG_PATH "/usr/local/opt/libffi/lib/pkgconfig"` in my fish, as suggested from running `brew info libffi` (which fixes the second issue I had). Otherwise, the build will not start.

It might be a good idea to include a note for this in the readme.

**Comment by CYBAI** [comment]
> This patch also works, but only if I `set -gx PKG_CONFIG_PATH "/usr/local/opt/libffi/lib/pkgconfig"` in my fish, as suggested from running `brew info libffi` (which fixes the second issue I had). Otherwise, the build will not start.
> 
> It might be a good idea to include a note for this in the readme.

Interesting. I just checked my `PKG_CONFIG_PATH` and it's `/usr/local/opt/expat/lib/pkgconfig`.

Maybe setting `PKG_CONFIG_PATH` is required?

Also, thanks for helping to check the patch works.

**Comment by Axighi** [comment]
I just pulled the master, still have this problem.
`pkg-config --modversion gstreamer-1.0` results 1.14.4

**Comment by CYBAI** [comment]
Could you check if there's any set `PKG_CONFIG_PATH` in your environment?

**Comment by Axighi** [comment]
> Could you check if there's any set `PKG_CONFIG_PATH` in your environment?

It's empty.

**Comment by Axighi** [comment]
> Maybe setting `PKG_CONFIG_PATH` is required?

Yes. It's required.

**Comment by luisbg** [comment]
I also had to do the `subprocess.check_output` to make it work with git master.

**Comment by mrwonko** [comment]
I had to `export PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig:/usr/local/lib/pkgconfig/"` to make it work, merely adding the libffi pkgconfig was insufficient…

**Comment by jdm** [comment]
Yes. Our readme calls that out, but we should fix #23177.

**Comment by o0Ignition0o** [comment]
Fwiw I just ran into the same issue, and did what @mrwonko suggested. It allowed me to start the build but I ran into an other error: 

```
error: failed to run custom build command for `gstreamer-sdp-sys v0.8.0`

Caused by:
  process didn't exit successfully: `/Users/ignition/Projects/mozilla/servo/target/debug/build/gstreamer-sdp-sys-1cd0ac0217628654/build-script-build` (exit code: 1)
--- stderr
`"pkg-config" "--libs" "--cflags" "gstreamer-sdp-1.0 >= 1.8"` did not exit successfully: exit code: 1
--- stderr
Package gstreamer-sdp-1.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gstreamer-sdp-1.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'gstreamer-sdp-1.0' found
```

I then did what the [Gstreamer install documentation says](https://gitlab.freedesktop.org/gstreamer/gstreamer-rs#homebrew) and it worked.

The path they provide is the following:

`export PKG_CONFIG_PATH="/Library/Frameworks/GStreamer.framework/Versions/Current/lib/pkgconfig${PKG_CONFIG_PATH:+:$PKG_CONFIG_PATH}"`
